osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -77,8 +77,9 @@ public IEnumerable<BeatmapSetInfo> Beatmaps
 
         private readonly List<Panel> panels = new List<Panel>();
 
-        private BeatmapGroup selectedGroup;
+        private readonly Stack<BeatmapGroup> randomSelectedBeatmaps = new Stack<BeatmapGroup>();

```

Comment:
Am thinking - how about storing this as a `Stack<KeyValuePair<BeatmapGroup, BeatmapPanel>>` and having `SelectPreviousRandom` work for panels too? At the moment if I filter e.g. AiAe, and F2, I can't go back.

osu!stable does random by panel as I suggest also.